mantra/representations/simplicial_connectivity.py

```diff
@@ -0,0 +1,300 @@
+from typing import Dict, List
+from networkx import incidence_matrix
+import torch
+import numpy as np
+from scipy.sparse import csr_matrix
+import scipy
+
+import torch_geometric.transforms as T
+from torch_geometric.data import Data
+
+from mantra.representations.simplex_trie import SimplexTrie, Simplex
+
+
+class AddSimplexTrie(T.BaseTransform):
+    """Add simplex trie to object.
+
+    Takes in a `Data` object and computes a `SimplexTrie` for it.
+    Then it adds it to it's storage to save up on computation
+    of connectivity matrices.
+
+
+    """
+
+    def forward(self, data: Data):
+        assert "triangulation" in data
+        simplex_trie = SimplexTrie()
+        for t in data.triangulation:
+            t_ = frozenset(t)
+            simplex_trie.insert(sorted(t_))
+        data.simplex_trie = simplex_trie
+        return data
+
+
+class ConnectivitySCTransform(T.BaseTransform):
```

Comment:
Can we find a better name for this? I would first suggest to write out "SC", i.e. "SimplicialComplex." I would also suggest to remove the `Transform` suffix, unless we need it. Since we have to import these transforms/representations, my understanding is that they only ever appear in a couple of places anyway, and I don't think the suffix is required.

This is **not** a good justification, but `pytorch-geometric` does not use this suffix very often (see [documentation](https://pytorch-geometric.readthedocs.io/en/stable/modules/transforms.html))